ModManager_Classes/Utils/ModCollectionHooks.cs

```diff
@@ -1,16 +1,31 @@
 ﻿using Imya.Models;
 using Imya.Models.Attributes;
 using Imya.Models.ModTweaker;
+using Imya.Utils.Validation;
 
 namespace Imya.Utils
 {
     public class ModCollectionHooks
     {
         ModCompabilityValidator compabilityValidator = new ModCompabilityValidator();
 
-        private ModCollectionHooks() {
-            ModCollection.Global!.ModAdded += UpdateWithTweak;
-            ModCollection.Global!.Updated += UpdateCompabilityCheck;
+        private ModCollectionHooks()
+        {
+            if (ModCollection.Global is null)
+                return; // TODO should not be possible, but make it noticable somehow
+
+            ModCollection.Global.ModAdded += UpdateWithTweak;
+            ModCollection.Global.ModAdded += ValidateOnAdd;
+            ModCollection.Global.Updated += UpdateCompabilityCheck;
+        }
+
+        private void ValidateOnAdd(Mod mod)
+        {
+            IModValidator[] validators = new[] { new ModContentValidator() };
```

Comment:
I think you need to press F5, that's still the old one without compatibility validator.

Reply:
ayy lmao I wrote this comment for the review before the latest commits. All good, I see now. 

Although, right now we are allocating everything over and over, why not setup the validators once on init?

Reply:
Habit of keeping things as local as possible and since the validators don't have members. But in this case a few lines above is still pretty close, so I move it up.